tests/test_readout.py

```diff
@@ -16,8 +19,17 @@
 from qadence.noise.protocols import Noise
 from qadence.operations import CNOT, RX, RZ, HamEvo, X, Y, Z
 from qadence.types import BackendName, ReadOutOptimization
+from scipy.sparse import csr_matrix
+from scipy.sparse.linalg import gmres
+from scipy.stats import wasserstein_distance
 
 from qadence_protocols.mitigations.protocols import Mitigations
+from qadence_protocols.mitigations.readout import (
+    matrix_inv,
+    mle_solve,
+    subspace_kron,
+    tensor_rank_mult,
+)
 
 
 @pytest.mark.flaky(max_runs=5)
```

Comment:
I think it does make sense to parametrise over the optimisation types so that all of them are checked and we can also assess the quality of the mitigations methods.

Reply:
Or loop over them in the test itself.

Reply:
Not all the protocols are meant for the same type of problem, but there is some more comparisons that can be done
I will move this to an issue and work on it after the strategies are put in place. 

[Add tests for comparisons of readout methods](https://github.com/pasqal-io/qadence-protocols/issues/20)